models/src/main/scala/coop/rchain/models/rholangn/BindingsFromProto.scala

```diff
@@ -0,0 +1,426 @@
+package coop.rchain.models.rholangn
+
+import coop.rchain.models.Connective.ConnectiveInstance._
+import coop.rchain.models.Expr.ExprInstance._
+import coop.rchain.models.GUnforgeable.UnfInstance._
+import coop.rchain.models.Var.VarInstance._
+import coop.rchain.models._
+import scalapb.GeneratedMessage
+
+import scala.annotation.unused
+
+private[rholangn] object BindingsFromProto {
+
+  def fromProto(p: Par): ParN = {
+    val terms: Seq[GeneratedMessage] =
+      Seq(p.sends, p.receives, p.news, p.exprs, p.matches, p.unforgeables, p.bundles, p.connectives)
+        .filter(_.nonEmpty)
+        .flatten
+    val ps: Seq[ParN] = terms.map(fromProtoMessage)
```

Comment:
In Denis code `terms` will be List[GeneratedMessages] and the `ps` will be  List[ParN], since immutable.Seq constructor delegates to List. So two collections will be created to my understanding. Thats why I suggested to use iterator specifically to have only one collection.

Reply:
@tgrospic already fixed it: https://github.com/RhonixLabs/rhonix/pull/34/commits/d3caa7bd5433686c59ece92535e8365eec291e9b